### PR TITLE
SAK-45573: Tests & Quizzes: Several datetimes in the Publish notifica…

### PR DESCRIPTION
**Pull request by hornersa**
…tion missing time zone

### CODE REVIEW

**Comment by ottenhoff** [comment]
Thanks for taking a swing at this Sean. I think the difficult part of this is that we do *not* want timezones displaying throughout the UI. Sakai should just be using the user's preferred time zone without having to explicitly label it. But yes, in emails, timezone info can be helpful especially if the user is not in the server timezone. 

After a quick glance at your PR, it seems like you are likely changing every display to use timezone info?

**Comment by hornersa** [comment]
Hello, Sam. Thanks for taking a look at this. The current calls for methods I've proposed to change are actually only in a few places, all of which pertain to publishing events. Here for instance are all references pertaining to getDueDateInClientTimezoneString or its associated property dueDateInClientTimezoneString (called via JSP files):

```
[shorner@mudan samigo]$ grep -r getDueDateInClientTimezoneString | grep -v "/target"
samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java:  public String getDueDateInClientTimezoneString() {
samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java:  public String getDueDateInClientTimezoneString() {
samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/PublishAssessmentListener.java:        assessmentSettings.getDueDateInClientTimezoneString(), assessmentSettings.getTimedHours(), assessmentSettings.getTimedMinutes(),

[shorner@mudan samigo]$ grep -r dueDateInClientTimezoneString | grep -v "/target"
samigo-app/src/webapp/jsf/author/publishAssessment.jsp:                 <f:param value="#{assessmentSettings.dueDateInClientTimezoneString}" />
samigo-app/src/webapp/jsf/author/publishAssessment.jsp:                         <f:param value="#{publishedSettings.dueDateInClientTimezoneString}" />

```

If however the plan is to expand the usage of getDueDateInClientTimezoneString (and the three other related methods), then we could add additional methods to the respective classes to serve the problems identified with the Publish event.

**Comment by hornersa** [comment]
I've revised my commit for SAK-45573 in order to incorporate a fix associated with T&Q republish events. The proposed fix now includes changes to RepublishAssessmentListener.java, and the test plan for SAK-45573 has new steps appended--  (steps 27-38)-- in order to test how T&Q assessments, when republished affect the dates in the Description of the associated Calendar event.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java [hunk 1]
```diff
@@ -1170,7 +1170,7 @@ public String getStartDateInClientTimezoneString() {
       return this.originalStartDateString;
     }
     else {
-      return userTimeService.dateTimeFormat(startDate, new ResourceLoader().getLocale(), DateFormat.MEDIUM);
+      return userTimeService.dateTimeFormat(startDate, new ResourceLoader().getLocale(), DateFormat.LONG);
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java:1173]: It's not required to create a new ResourceLoader constantly use one that is already declared.

Comment (ottenhoff) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java]: this is resolved

**Pull request merged**